docs/framework_i18n.rst

```diff
@@ -7,9 +7,54 @@
 Internationalization Framework
 ==============================
 
------------
-Basic Usage
------------
+--------
+Tutorial
+--------
+
+The Red framework supports translations into different languages, making it easy 
+for translators to contribute new language files to the Red project, and to Cogs
+that use the framework. The framework is based on the popular gettext format, 
+(with some modifications), which is the standard for providing translations in 
+lots of different software projects.
+
+* ``.pot`` files are typically generated by the original author of the cog, and 
+  contain the basic messages in en_US (the language Red is written in), and is 
+  used as a template for translators to build their translations on.
+* ``.po`` files (portable objects) are language specific, for example 
+  ``en-GB.po`` for British English, or ``en-ES.po`` for Spanish. Note that Red 
+  uses a ``-`` character in the filename, not a ``_`` which is often seen as as 
+  the standard.
+* ``locales/`` directory, where ``.po`` files are stored.
+
+So, if your are the author of a Cog, or you are adding translations for the first
+time to python code, you would probably generate ``.pot`` files to make your code 
+easily translatable. See below for instructions.
+
+If you are trying to translate part of Red, or a Cog, you would write a ``.po`` 
+file.
+
+A technical detail, is that Red actually uses a slightly modified version of 
+``pygettext``, called ``redgettext`` (https://github.com/Cog-Creators/redgettext). 
+This is mostly just an implementation detail though and does not affect Cog 
+authors or translators. If you are used to gettext, it is important to explain 
```

Comment:
It does affect Cog authors in that our tool can extract command docstrings (which *are* part of cog's UI) while not extracting all the other docstrings (which generally aren't user-facing). This is what the `redgettext` usage example listed in this document does and is something that cannot be done with `pygettext` (which can only do no docstrings or all docstrings). The distinction probably isn't important for translators who just translate whatever `.pot` file is given to them but is definitely something that Cog authors would want to know.

More so, `pygettext` will sometime in the future supports more features than `pygettext` which is actually rather limited - it has no support for plurals *or* for context-dependent translations. Those features are already implemented in `redgettext` (see Cog-Creators/redgettext#9, Cog-Creators/redgettext#10, Cog-Creators/redgettext#11) and just haven't been released yet as I was hoping to have a working implementation of relevant runtime features in Red as well to make sure it aligns with what the support was implemented for.

While upstreaming those features to pygettext *was* mentioned as a possibility (https://discuss.python.org/t/modernize-and-add-missing-features-to-pygettext/26455), a year passed since so it's unclear that there's enough interest to get it there any time soon.